### ISSUE TITLE
Camera method to see if an area is visible

### ISSUE THREAD

**Issue by JMS55** [bevyengine/bevy#2644]
## What problem does this solve or what need does it fill?

I only want to play an animation if it would be visible. 

## What solution would you like?

Given an OrthographicCameraBundle, I want to know if a given rectangular area is on screen or not.

**Comment by JMS55** [comment]
Such a method might look like this, if you implemented it on `World`, using the first camera found:
```rust
fn is_rect_visible(&mut self, rect: Rect<f32>) -> bool {
    let camera = self
        .query::<(&OrthographicProjection, &GlobalTransform)>()
        .iter(self)
        .next();
    match camera {
        Some((projection, transform)) => {
            let p1 = Vec2::new(
                projection.left + transform.translation.x,
                projection.top + transform.translation.y,
            );
            let p2 = Vec2::new(
                projection.right + transform.translation.x,
                projection.bottom + transform.translation.y,
            );
            let p3 = Vec2::new(rect.left, rect.top);
            let p4 = Vec2::new(rect.right, rect.bottom);
            p2.y <= p3.y && p1.y >= p4.y && p2.x >= p3.x && p1.x <= p4.x
        }
        None => false,
    }
}
```

**Comment by mockersf** [comment]
I'm guessing you are in 2D? In which case there is an experimental frustum culling feature that you can enable:
https://github.com/bevyengine/bevy/blob/b13472dae495ecc7d1a80cbe1ad12718c0de9262/examples/2d/many_sprites.rs#L21-L24

It should add a component [`OutsideFrustum`](https://docs.rs/bevy/0.5.0/bevy/render/draw/struct.OutsideFrustum.html) to sprite that are outside camera

**Comment by JMS55** [comment]
Yes I am in 2D. That doesn't work for me however - If something is moving from offscreen to onscreen, that won't give me the info I need to know that I should animate it, as the player would be able to see the animation (and to then pause other things while it happens).

**Comment by alice-i-cecile** [comment]
I think this would work well as a method on the relevant projections.

**Comment by ZainlessBrombie** [comment]
I think multiplying the Coordinates to be checked with the Camera matrix and checking if they intersect with the view frustrum would be the most straightforward way to go about this

**Comment by wilk10** [comment]
Can this be closed now that `ComputedVisibility` has been added to `SpriteBundle` and `SpriteSheetBundle`?

**Comment by alice-i-cecile** [comment]
IMO no: that's useful and related but not a general purpose API that allows people to build their own logic here.

**Comment by BenjaminBrienen** [comment]
Double checking that this is still actionable and trivial

**Comment by richchurcher** [comment]
Heh. You read my mind, @BenjaminBrienen! Was about to ask the same. You take this one tho :)
